Minigames/src/main/java/au/com/mineauz/minigames/managers/PlaceHolderManager.java

```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Override
+    public String onRequest(OfflinePlayer p, String params) {
+        return super.onRequest(p, params);
+    }
+
+    @Override
+    public boolean canRegister(){
+        return true;
+    }
+
+    @Override
+    public boolean persist(){
+        return true;
+    }
+    @Override
+    public String getIdentifier() {
+        return Minigames.getPlugin().getName();
+    }
+
+    @Override
+    public String getAuthor() {
+        return Minigames.getPlugin().getDescription().getAuthors().toString();
+    }
+
+    @Override
+    public String getVersion() {
+        return Minigames.getVERSION().toString();
+    }
+
+    @Override
+    public String onPlaceholderRequest(Player player, String identifier){
+
+        if(player == null){
+            return "";
+        }
+        Set<String> games = plugin.getMinigameManager().getAllMinigames().keySet();
+        if(identifier.contains("_")){
+            String[] parts = identifier.split("_");
+            String gameName = parts[0];
+            if(games.contains(gameName)){
+                Minigame minigame = plugin.getMinigameManager().getMinigame(gameName);
+                try {
+                    switch (parts[1]) {
+                        case "enabled":
+                            return Boolean.toString(minigame.isEnabled());
+                        case "maxPlayers":
+                            return Integer.toString(minigame.getMaxPlayers());
+                        case "currentPlayers":
+                            return Integer.toString(minigame.getPlayers().size());
+                        case "type":
+                            return minigame.getType().getName();
+                        case "mechanic":
+                            return minigame.getMechanicName();
+                        case "state":
+                            return minigame.getState().name();
+                        case "objective":
+                            return minigame.getObjective();
+                        case "gameType":
+                            return minigame.getGametypeName();
+                        case "timeLeft":
+                            return Integer.toString(minigame.getMinigameTimer().getTimeLeft());
+                        case "name":
+                            return minigame.getName(true);
+                        default:
+                            return null;
+                    }
+                }catch (Exception e){
+                    plugin.getLogger().warning("Error processing PAPI:" + identifier);
+                    plugin.getLogger().warning(e.getMessage());
+                    if(plugin.isDebugging()){
+                        e.printStackTrace();
+                    }
+                    return null;
+                }
+            } else {
+                //this means the first part is not a gameName ?? what else could it be
+                return null;
+            }
+        }
```

Comment:
Suggestion: Turn this into an `else` because you only want to do the case below if the `contains("_")` fails